### ISSUE TITLE
npm ERR! ERESOLVE unable to resolve dependency tree

### ISSUE THREAD

**Issue by spacecat** [NourAlzway/use-router-with-events#7]
```
    ~/r/use-router-with-events-01  on   main  npm i use-router-with-events                                         ✔  at 17:35:31  
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: use-router-with-events-01@0.1.0
npm ERR! Found: react@19.0.0-rc-65a56d0e-20241020
npm ERR! node_modules/react
npm ERR!   react@"19.0.0-rc-65a56d0e-20241020" from the root project
npm ERR!   peer react@"^18.2.0 || 19.0.0-rc-65a56d0e-20241020" from next@15.0.0
npm ERR!   node_modules/next
npm ERR!     next@"15.0.0" from the root project
npm ERR!     peer next@">=14.0.0" from use-router-with-events@0.1.3
npm ERR!     node_modules/use-router-with-events
npm ERR!       use-router-with-events@"*" from the root project
npm ERR!   1 more (react-dom)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@">=16.8.0" from use-router-with-events@0.1.3
npm ERR! node_modules/use-router-with-events
npm ERR!   use-router-with-events@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/xyz/.npm/_logs/2024-10-22T15_36_24_710Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/xyz/.npm/_logs/2024-10-22T15_36_24_710Z-debug-0.log
```

Repro steps:

1. `npx create-next-app@latest`
2. `npm i use-router-with-events`
3. The error message from above appears.

I'm using node.js version `20.9.0` and npm version `10.1.0`

**Comment by NourAlzway** [comment]
Hi @spacecat,

Thanks for reporting this issue!

After reviewing I found that the error occurs because `use-router-with-events@0.1.3` didn’t fully support the `-rc` (release candidate) versions in its `peerDependencies`, which Next.js 15 uses. I’ve just published a new version, `0.2.0`, which explicitly includes support for React release candidates.

### Solution
To resolve this issue, please try updating to `use-router-with-events@0.2.0`:

```bash
npm install use-router-with-events@0.2.0
```

**Context:** Next.js 15 specifies `react@"^18.2.0 || 19.0.0-rc"` in its dependencies. Prior versions of `use-router-with-events` expected only stable versions, causing a conflict with the `-rc` tag Next.js 15 introduces. Version `0.2.0` of `use-router-with-events` now accommodates these `-rc` patterns, so updating should address the conflict.

Thank you again for your feedback, and feel free to reopen this issue if further help is needed.

Best regards, 
Nour

**Comment by spacecat** [comment]
Thanks for the time and effort.
A lot of stuff happening at vercel ATM.
Noticed the major version update too along with the update to React 19 RC.

Anyways, I'm not using this library anymore since it does not fulfill the requirements for my app but still great package and I'm sure others will find it useful.
